### DISCUSSION TITLE
Fail to import existing database file using `IDBMirrorVFS`

### DISCUSSION THREAD

**Discussion by subframe7536** [rhashimoto/wa-sqlite#232] [Q&A] [answered]
I just fork the file I/O demo and [change the VFS to `IDBMirrorVFS`](https://github.com/subframe7536/wa-sqlite/commit/4da59f3e6b5f9ce0ba8ca059d5c848afd8b36796) and it prompts error (Seems its `txActive` is `null`):

![image](https://github.com/user-attachments/assets/d4ba77d9-2ad6-4b18-9b5f-76e821315064)

Also, in my local test, skip execute `jTruncate` will suppress the error, but also cannot directly read the updated db data, reopen it will make it works.

**Comment by rhashimoto** [comment, marked as answer]
Thanks for the report. This is a [bug in the VFS](https://github.com/rhashimoto/wa-sqlite/issues/233). Since you have a fork, you can try [this commit](https://github.com/rhashimoto/wa-sqlite/commit/84a6c1e10d8540baab70c469d6896d3dc7fd01c5). It may be a few weeks until I can work on merging a real PR.

**Comment by subframe7536** [comment]
Thanks! With this commit, the database is imported, but cannot been verified:

![image](https://github.com/user-attachments/assets/1f81d66c-0895-4ee2-a69a-cb0ccdcc2303)

**Reply by rhashimoto** [reply]
You have to use the same VFS in the verifier.

**Reply by subframe7536** [reply]
Thanks, that works!
